src/Bicep.Core/Configuration/ConfigurationManager.cs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Abstractions;
+using System.Reflection;
+
+namespace Bicep.Core.Configuration
+{
+    public class ConfigurationManager : IConfigurationManager
+    {
+        public const string BuiltInConfigurationResourceName = "Bicep.Core.Configuration.bicepconfig.json";
+
+        private static readonly IConfiguration builtInRawConfiguration = BuildBuiltInRawConfiguration();
+
+        private static readonly Lazy<RootConfiguration> builtInConfigurationLazy =
+            new(() => RootConfiguration.Bind(builtInRawConfiguration, BuiltInConfigurationResourceName));
+
+        private static readonly Lazy<RootConfiguration> builtInConfigurationWithAnalyzersDisabledLazy =
+            new(() => RootConfiguration.Bind(builtInRawConfiguration, BuiltInConfigurationResourceName, disableAnalyzers: true));
+
+        private readonly IFileSystem fileSystem;
+
+        public ConfigurationManager(IFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+        }
+
+        public RootConfiguration GetBuiltInConfiguration(bool disableAnalyzers = false) => disableAnalyzers
+            ? builtInConfigurationWithAnalyzersDisabledLazy.Value
+            : builtInConfigurationLazy.Value;
```

Comment:
I thought we'd pulled out support for setting `disableAnalyzers` entirely. Do you know which code paths this is being used in?

Reply:
Yup. I thought about removing this as well but later realized that this is needed to fix #4471. If a custom bicepconfig.json file is invalid we will fallback to the default configuration with linter disabled (see https://github.com/Azure/bicep/pull/4576/files#diff-3707a23954e587bd8ed9df408b7c7990474a7bf1765d48e00b6930516ad39c28R299-R300).

Reply:
Wouldn't we just want to fall back to the **default** default configuration in this scenario?

Reply:
One justification is that if the user have some default linter rules disabled in their broken bicepconfig.json, they will get warnings for those rules which could be confusing.